processing/src/main/java/io/druid/segment/incremental/IncrementalIndexStorageAdapter.java

```diff
@@ -532,14 +530,13 @@ public Long get()
                 final Integer metricIndexInt = index.getMetricIndex(column);
                 if (metricIndexInt != null) {
                   final int metricIndex = metricIndexInt;
-
-                  final ComplexMetricSerde serde = ComplexMetrics.getSerdeForType(index.getMetricType(column));
+                  final Class classOfObject = index.getMetricClass(column);
```

Comment:
are those changes required? If we want to get this into 0.9.1 I'd rather we piggy back on the existing mechanisms and make the changes in 0.9.2 to avoid breaking things in unexpected ways